app/models/provider.rb

```diff
@@ -79,6 +79,8 @@ class Provider < ApplicationRecord
                     },
                   }
 
+  scope :with_active_hei, -> { kept.where(accredited: true).where("accreditation_id ~ ?", "^[1][0-9]{3}") }
```

Comment:
Couple of questions here:
1. Do we need the `with` in the name? I think a Provider _is/isn't_ an HEI, rather than being _with_ an HEI.
2. Do we need to check four characters here, or just the first? If we want to make sure the `accreditation_id` matches our expectations of a 4 digit number then we might want a `$` at the end too?
```suggestion
  scope :active_hei, -> { kept.where(accredited: true).where("accreditation_id ~ ?", "^1[0-9]{3}$") }
```

Reply:
Fixed in https://github.com/DFE-Digital/register-trainee-teachers/pull/5191/commits/9ec379eb548bfcbe36e2e1dfe6a219982e83cf66